WcaOnRails/app/models/competition.rb

```diff
@@ -233,6 +234,7 @@ def update_foreign_keys
       CompetitionDelegate,
       CompetitionOrganizer,
       DelegateReport,
+      CompetitionTab,
     ].each do |model|
       foreign_key = model.column_names.include?("competitionId") ? "competitionId" : "competition_id"
       model.where(foreign_key => id_was).update_all(foreign_key => id)
```

Comment:
I didn't mean to say that's a deal breaker. Just analyzing design. This solves 98% of the problem, and the problem won't be around for long, so it's good enough.

My approach would be to just list the current tables in the production code, combined with a test that finds all columns named `competition_id` or `competitionId` in the schema, and goes red if it finds an unlisted one. That only fails if people commit with failing tests or disable the test. Or name the column something weird.

Code it up if you want, but I'm happy with what you have.